login/service.go

```
@@ -648,8 +650,20 @@ func (keycloak *KeycloakOAuthProvider) CreateOrUpdateKeycloakUser(accessToken st
 		if !approved {
 			return nil, nil, autherrors.NewUnauthorizedError(fmt.Sprintf("user '%s' is not approved", claims.Username))
 		}
-		user = new(account.User)
-		identity = &account.Identity{}
+		isNewUser := true
+		user, identity, err = keycloak.fetchUserInfoFromRemoteService(ctx, WITEndpointUserProfile, &accessToken)
+		if user != nil && identity != nil {
```

Make sure you don't ignore errors other than Not Found.